File: Exercises/C++/Dietel & Dietel C++ How to Program/Chapter_3/Examples/Validation/Validation/GradeBook.cpp
// Fig. 3.16: GradeBook.cpp
// Implementations of the GradeBook member-function definitions
// The setCourseName function performs validation
#include <iostream>
#include "GradeBook.h" // include definition of class GradeBook
using namespace std;

// constructor initializes courseName with string supplied as argument
GradeBook::GradeBook( string name )
{
	setCourseName( name ); // validate and store courseName
} // end GradeBook constructor

// function that sets the course name;
// ensures that the course name has at most 25 characters
void GradeBook::setCourseName( string name )
{
	if ( name.length() <= 25) // if name has 25 or fewer characters
		courseName = name; // store the course name in the object

	if (name.length() > 25) // if name has more than 25 characters
	{
		// set courseName to first 25 characters of parameter name
		courseName = name.substr( 0, 25 ); //start at 0, length of 25

		cout << "Name \"" << name << "\" exceeds maximum length (25). \n" << "Limiting courseName to first 25 characters.\n" << endl;
	} // end if

} //end function setCourseName

// function to get course name
string GradeBook::getCourseName()
{
	return courseName;	// return object's courseName
} // end function getCourseName

// display a welcome message to the GradeBook user
void GradeBook::displayMessage()
{
	// call getCourseName to get the courseName
	cout << "Welcome to the grade book for\n" << getCourseName() << "!" << endl;
} // end function displayMessage

File: Exercises/Java/Java_How to Program_8th Edition/Visual Studio/Chapter 3_Introduction to Classes and Objects/Examples/GradeBook3/GradeBook3/src/main/java/pkg/GradeBook.java
package pkg;

// Fig. 3.7: GradeBook.java
// GradeBook class that contains a courseName instance variable
// and methods to set and get its value.

public class GradeBook 
{
	private String courseName; // course name for this GradeBook

	// method to set the course name
	public void setCourseName(String name)
	{
		courseName = name; // store the course name
	} // end method setCourseName

	// method to retrieve the course name
	public String getCourseName()
	{
		return courseName;
	}

	// display a welcome message to the GradeBoook user
	public void displayMessage()
	{
		// calls getCourseName to get the name of
		// the course this GradeBook represents
		System.out.printf("Welcome to the grade book for\n%s!\n",
			getCourseName());
	} // end method displayMessage

} // end class GradeBook


